### PR TITLE
Module 9, Unit testing

### PR DESCRIPTION
**Pull request by Nastya-Korsak**
Module 9, Unit testing

### CODE REVIEW

File: Module_9/HarryPotter/Bookstore.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+
+            Dictionary<Books, int> harryPotterBooks = GetHarryPotterBooksCount(books, ref otherBooks);
+
+            double finallyPrice = 0;
+
+            finallyPrice += otherBooks * 8;
+
+            while (harryPotterBooks.All(v => v.Value > 0))
+            {
+                finallyPrice += (5.0 * 8.0) - (5.0 * 8.0 * 0.25);
```
Comment (alexander-trishin) [inline Module_9/HarryPotter/Bookstore.cs:25]: Try to avoid "magic numbers". Instead use constants

File: Module_9/HarryPotter/Bookstore.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                foreach (var v in harryPotterBooks)
+                {
+                    if (harryPotterBooks[v.Key] > 0)
+                    {
+                        harryPotterBooks[v.Key] -= 1;
+                    }
+                }
+            }
+
+            while (harryPotterBooks.Count(v => v.Value > 0) == 2)
```
Comment (alexander-trishin) [inline Module_9/HarryPotter/Bookstore.cs:56]: 3 duplicate blocks. You could just calculate coefficients and perform just 1 cycle.

File: Module_9/HarryPotter/Bookstore.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+                    harryPotterBooks[Books.HarryPotter4] += 1;
+                }
+                else if (book == Books.HarryPotter5)
+                {
+                    harryPotterBooks[Books.HarryPotter5] += 1;
+                }
+                else
+                {
+                    otherBooks += 1;
+                }
```
Comment (alexander-trishin) [inline Module_9/HarryPotter/Bookstore.cs:113]:
```suggestion
                if (harryPotterBooks.ContainsKey(book))
                {
                    harryPotterBooks[book] += 1;
                }
                else
                {
                    otherBooks += 1;
                }
```

File: Module_9/MineFields/MFSings.cs [hunk 4]
```diff
@@ -0,0 +1,8 @@
+﻿namespace MineFields
+{
+    public enum MFSings
```
Comment (alexander-trishin) [inline Module_9/MineFields/MFSings.cs:3]:
```suggestion
    public enum MineFieldSign
```

**Review by alexander-trishin — APPROVED** [review]
Hi Nastya,

In general, everything is okay. But some parts could be simplified.

File: Module_9/HarryPotter/IBookstore.cs [hunk 5]
```diff
@@ -0,0 +1,9 @@
+﻿using System.Collections.Generic;
+
+namespace HarryPotter
+{
+    public interface IBookstore
+    {
+        public double GetFinalPrice(List<Books> books);
```
[the remark this answers is not in the document]
Comment (Nastya-Korsak) [reply Module_9/HarryPotter/IBookstore.cs]: fixed

File: Module_9/FizzBuzz/FizzBuzz.cs [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+
+            for (int i = 1; i <= 100; i++)
+            {
+                listOfHandledNumbers.Add(HandleNumber(i));
+            }
+
+            return listOfHandledNumbers;
+        }
+
+        private bool IsNumberDivisibleBy3(int number)
```
[the remark this answers is not in the document]
Comment (Nastya-Korsak) [reply Module_9/FizzBuzz/FizzBuzz.cs]: fixed

File: Module_9/MineFields/FieldHelper.cs [hunk 7]
```diff
@@ -0,0 +1,15 @@
+﻿namespace MineFields
+{
+    public static class FieldHelper<T>
+    {
+        public static int GetRowsNumber(T[,] field)
+        {
+            return field.GetUpperBound(0) + 1;
```
[the remark this answers is not in the document]
Comment (Nastya-Korsak) [reply Module_9/MineFields/FieldHelper.cs:12]: fixed

File: Module_9/HarryPotter/Bookstore.cs (hunk 1 above)
Comment (Nastya-Korsak) [reply Module_9/HarryPotter/Bookstore.cs]: fixed

File: Module_9/HarryPotter/Bookstore.cs (hunk 2 above)
Comment (Nastya-Korsak) [reply Module_9/HarryPotter/Bookstore.cs]: fixed

File: Module_9/HarryPotter/Bookstore.cs (hunk 3 above)
Comment (Nastya-Korsak) [reply Module_9/HarryPotter/Bookstore.cs]: fixed

File: Module_9/MineFields/MFSings.cs (hunk 4 above)
Comment (Nastya-Korsak) [reply Module_9/MineFields/MFSings.cs]: fixed

**Pull request merged**